ui/app/controllers/jobs/index.js

```diff
@@ -2,9 +2,28 @@ import { inject as service } from '@ember/service';
 import { alias } from '@ember/object/computed';
 import Controller, { inject as controller } from '@ember/controller';
 import { computed } from '@ember/object';
+import { scheduleOnce } from '@ember/runloop';
+import intersection from 'lodash.intersection';
 import Sortable from 'nomad-ui/mixins/sortable';
 import Searchable from 'nomad-ui/mixins/searchable';
 
+// An unattractive but robust way to encode query params
+const qpSerialize = arr => (arr.length ? JSON.stringify(arr) : '');
+const qpDeserialize = str => {
+  try {
+    return JSON.parse(str)
+      .compact()
+      .without('');
+  } catch (e) {
+    return [];
+  }
+};
+
```

Comment:
Hehe I see what you mean here by unattractive! So my URL right now has:

```
?status=%5B"pending"%2C"running"%2C"dead"%5D
```

Generally I've seen these things done with `status[]=thing`, I'm not sure how ember treats that though, I did notice something the other day related to this, I'll try and dig it out again in a bit incase it helps.

I'm not entirely sure that `"` should be in URLs, and also that `,` shouldn't - probably worth having a look to see what is 'usual' here. Do you need the `"`'s? Are you expecting to have `true` and `"true"` at the same time?

Anyway, if you take a look lemme know what you find out.

Reply:
Here's the thing I was looking at the other day which is ever so slightly related to this

https://github.com/tildeio/route-recognizer/blob/0940966757104ea5297717102b1ae3dc260ee8ee/lib/route-recognizer.ts#L564-L574

Looks like ember is aware of the `key[]=value` thing, just be careful as I'm not sure it encodes keys properly when you do that. I don't think that would effect you here anyway.

Reply:
I was honestly expecting Ember to use the `status[]=&status[]=` style when I bound an array to a query param, but that's not what happened. Instead it basically does what I'm doing now: JSON stringifying.

As for encoding characters, it's curious that you are seeing `"` and `,` in the URL, what browser are you using? In Chrome, they were automatically encoded. Not sure if that's a property of Ember or Chrome.

> Do you need the "'s? Are you expecting to have true and "true" at the same time?

Yes-ish. Ideally I would change `qpSerialize` to just `arr.join(',')`, but commas (and basically everything) is a legal job name and datacenter character. So there are other possible encoding schemes than `JSON.stringify`, but it's not worth spinning my wheels on that.

Reply:
Definitely getting `"`s in Chrome

![screenshot 2019-02-05 at 16 34 08](https://user-images.githubusercontent.com/554604/52288364-f460da80-2963-11e9-89e3-0ef905e51721.png)